tests/tools/archive/test_complex.py

```diff
@@ -196,3 +197,92 @@ def get_hash_from_db_content(grouplabel):
+    nodes = []
+    for i, user in enumerate(users):
+        node = orm.Int(i, user=user)
+        node.label = f'node_{i}'
+        node.store()
+        nodes.append(node)
+
+    # Create some links between nodes
+    calc_nodes = []
+    for i in range(min(3, nb_entities)):
+        calc = orm.CalculationNode()
+        calc.base.links.add_incoming(nodes[i], LinkType.INPUT_CALC, f'input_{i}')
+        calc.store()
+        calc.seal()
+        calc_nodes.append(calc)
+        nodes.append(calc)
+
+    # Export with small filter_size to force batching in multiple functions
+    export_file = tmp_path / 'large_export.aiida'
+    create_archive(nodes, filename=export_file, filter_size=3)
+
+    # Verify by importing
+    aiida_profile_clean.reset_storage()
+    import_archive(export_file)
+
+    # Check all entities were properly exported/imported
+    assert orm.QueryBuilder().append(orm.Node).count() == len(nodes)
+    assert orm.QueryBuilder().append(orm.User).count() == nb_entities + 1  # +1 for default user
+
+    # Assert for calculation nodes specifically
+    calc_node_count = orm.QueryBuilder().append(orm.CalculationNode).count()
+    assert calc_node_count == len(calc_nodes)
+
+    # Assert for data nodes specifically
+    data_node_count = orm.QueryBuilder().append(orm.Int).count()
+    assert data_node_count == nb_entities
+
+    # Verify links were preserved
+    link_count = orm.QueryBuilder().append(entity_type='link').count()
+    assert link_count >= 3  # At least the links we created
+
+
+@pytest.mark.timeout(600)  # Should finish in ~350s
+@pytest.mark.nightly
+@pytest.mark.usefixtures('aiida_profile_clean')
+def test_large_archive_export_operr_regression(pytestconfig, tmp_path, create_int_nodes):
+    """Regression test: ensure OperationalError occurs (not) with too large (sufficiently small) filter_size."""
+
+    from sqlalchemy.exc import OperationalError
+
+    num_nodes = 66_000  # Minimum value that OpErr is raised
+
+    _ = create_int_nodes(num_nodes)
+
+    # NOTE: Need to obtain a selection of `orm.Node`s, otherwise, with `entities=None`, where the profile is exported in
+    # full, no QB filters are applied, as the `_collect_all_entities` method is called
+    orm_nodes = orm.QueryBuilder().append(orm.Node).all(flat=True)
+
+    # NOTE: This passes for both backends
+    export_file = tmp_path / 'should_work.aiida'
+    create_archive(entities=orm_nodes, filename=export_file, filter_size=1000)
+
+    # NOTE: Run the failing test at the end as the exception leaves sqlachemy in an erroneus state,
+    # so the `orm_nodes` list cannot be re-used. Trying to do so gives:
+    # sqlalchemy.orm.exc.DetachedInstanceError: Instance <DbNode at 0x75d3826f1b40> is not bound to a Session;
+    # attribute refresh operation cannot proceed (Background on this error at: https://sqlalche.me/e/20/bhk3)
+    # .venv/lib/python3.10/site-packages/sqlalchemy/orm/loading.py:1603: DetachedInstanceError
+
```

Comment:
This is a bit concerning. My worry here is whether it could affect other tests that would run after this one. 
Could you test it? You can mark this test to run first in the file (something like `@pytest.mark.first` or some such, check docs).